### PR TITLE
chore: enable ESLint on `test` folder

### PR DESCRIPTION
**Pull request by LekoArts**
### Summary

Enables ESLint for our tests and adds `eslint-plugin-jest`

### Relevant links (GitHub issues, Notion docs, etc.) or a picture of cute animal

Fixes https://github.com/netlify/pod-ecosystem-frameworks/issues/466

### Standard checks:

- [ ] Check the Deploy Preview's Demo site for your PR's functionality
- [ ] Add docs when necessary

---

🧪 Once merged, make sure to update the version if needed and that it was published correctly.

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
 demos
 packages/runtime/src/templates/edge
 packages/runtime/src/templates/edge-shared
 packages/runtime/lib
 packages/runtime/dist-types
-jestSetup.js
\ No newline at end of file
+jestSetup.js
+test/e2e
+test/fixtures/broken_next_config/next.config.js
```
Comment (nickytonline) [inline .eslintignore:11]: [dust] since we're updating this add a new line that's missing:

File: test/helpers/config.spec.ts [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
 
-const netlifyConfig = { build: { command: 'npm run build' }, functions: {}, redirects: [], headers: [] } as NetlifyPluginOptions["netlifyConfig"]
+import { generateCustomHeaders, NextConfig } from '../../packages/runtime/src/helpers/config'
+
+const netlifyConfig = {
+  build: { command: 'npm run build' },
+  functions: {},
+  redirects: [],
+  headers: [],
+} as NetlifyPluginOptions['netlifyConfig']
```
Comment (nickytonline) [inline test/helpers/config.spec.ts:10]: Maybe for another PR, but prefer not to cast. Add anything missing if required.

```suggestion
const netlifyConfig: NetlifyPluginOptions['netlifyConfig'] = {
  build: { command: 'npm run build' },
  functions: {},
  redirects: [],
  headers: [],
}
```

File: test/helpers/files.spec.ts [hunk 3]
```diff
@@ -63,7 +59,7 @@ const REWRITES: Rewrites = [
 ]
 
 describe('files utility functions', () => {
-  test('middleware tester matches correct paths', () => {
+  it('middleware tester matches correct paths', () => {
```
Comment (nickytonline) [inline test/helpers/files.spec.ts:62]: Is `it` what we use everywhere else? I can't remember between all the projects we have. 😅

**Review by nickytonline — APPROVED** [review]
Looks good @LekoArts! Left some comments, but nothing blocking. :shipit:

File: test/helpers/files.spec.ts (hunk 3 above)
Comment (LekoArts) [reply test/helpers/files.spec.ts:62]: I've seen `it` used in e.g. Gatsby plugin, too

File: test/helpers/config.spec.ts (hunk 2 above)
Comment (LekoArts) [reply test/helpers/config.spec.ts:10]: It seems that for the sake of the test, those key/values are enough, a lot of stuff is missing and it just unblocked things

**Pull request merged**